app/assets/javascripts/search-vocab.js

```diff
@@ -0,0 +1,56 @@
+$(document).ready(function() {
+  $('.add-to-vocab-btn').click(function(e) {
```

Comment:
Oh one last thing! I think its probably worth wrapping the all the logic/functions here in a initialization function and then calling that function at the bottom but **only** if the .add-to-vocab-btn element is present (or some sort of similar presence check). That way we can stop it from trying to run all of this on pages that don't need the functionality or don't feature the class which could potentially throw some console errors.